### PR TITLE
Add support for QUEST_OBJECTIVE_CRITERIA_TREE

### PR DESCRIPTION
**Pull request by rstebler**
[//]: # (***************************)
[//]: # (** FILL IN THIS TEMPLATE **)
[//]: # (***************************)

**Changes proposed:**

-  Add support for QUEST_OBJECTIVE_CRITERIA_TREE

**Target branch(es):** 3.3.5/master

- [ ] 3.3.5
- [x] master

**Tests performed:** (Does it build, tested in-game, etc.)
- [x] Does it build?
- [x] Tested in-game? Tested with quest 38312

### CODE REVIEW

File: src/server/game/Achievements/AchievementMgr.cpp [hunk 1]
```diff
@@ -65,7 +65,7 @@ bool AchievementMgr::CanUpdateCriteriaTree(Criteria const* criteria, CriteriaTre
 {
     AchievementEntry const* achievement = tree->Achievement;
     if (!achievement)
-        return false;
+        return true;
```
Comment (Krudor) [inline src/server/game/Achievements/AchievementMgr.cpp]: What?
If you can't find an achievement linked to the criteriatree, return true no matter what? This is really really bad.

Comment (rstebler) [inline src/server/game/Achievements/AchievementMgr.cpp]: You are right. Sorry I'm new to TC and still learning how everything works :)
What about the CriteriaHandler check in the new commit?

Comment (Krudor) [inline src/server/game/Achievements/AchievementMgr.cpp]: If it's not an achievement then it's not supposed to update it, this is the achievement manager, it wasn't designed to handle quest's criteria trees.
If there's another system that starts using criterias then I guess we need a NEW manager for it.

Comment (Krudor) [inline src/server/game/Achievements/AchievementMgr.cpp]: Take a look at Player::UpdateCriteria, it sends updates to the achievement manager, scenario, and guild system, which are all separate criteria handlers.

Comment (rstebler) [inline src/server/game/Achievements/AchievementMgr.cpp]: Yea a new manager for that would be the best thing. If I have time I will have a look at it but I think it's best to leave this work to you and other more experienced people than me :)
Thanks for your time.

**Comment by Aokromes** [comment]
If no one have opposition i will merge this tomorrow.

File: src/server/game/Achievements/AchievementMgr.cpp (hunk 1 above)
Comment (jackpoz) [inline src/server/game/Achievements/AchievementMgr.cpp]: This point still needs to be addressed. Post here to reopen the PR once done

Comment (rstebler) [inline src/server/game/Achievements/AchievementMgr.cpp]: This correction needs an additional check? 6688ad03d35bb264e41f5fbfcc5cee377c84d913

Comment (jackpoz) [inline src/server/game/Achievements/AchievementMgr.cpp]: @Krudor can you check if the changes are ok ?

**Comment by Aokromes** [comment]
btw @rstebler i sugest you to jump irc irc.rizon.net #trinity

**Pull request closed without merging**